plugins/3MFReader/ThreeMFReader.py

```diff
@@ -120,8 +122,8 @@ def _convertSavitarNodeToUMNode(self, savitar_node):
                     um_node.callDecoration("setActiveExtruder", default_stack.getId())
 
                 # Get the definition & set it
-                definition = QualityManager.getInstance().getParentMachineDefinition(global_container_stack.getBottom())
-                um_node.callDecoration("getStack").getTop().setDefinition(definition.getId())
+                definition_id = getMachineDefinitionIDForQualitySearch(global_container_stack.definition)
+                um_node.callDecoration("getStack").getTop().setDefinition(definition_id)
```

Comment:
Use _stack.userChanges_

Reply:
Changed in ccd4bb285

Reply:
This commit is reverted in f304e7f2e73828c4d2f1c5ceeb15c3f6196bbc66 because those `PerObjectContainerStack`s are not `CuraContainerStack`s, so we cannot use properties such as `userChanges`.